ompi/request/request.h

```diff
@@ -532,7 +532,7 @@ static inline int ompi_request_complete(ompi_request_t* request, bool with_signa
 
             ompi_wait_sync_t *tmp_sync = (ompi_wait_sync_t *) OPAL_ATOMIC_SWAP_PTR(&request->req_complete,
                                                                                     REQUEST_COMPLETED);
-            if( REQUEST_PENDING != tmp_sync ) {
+            if (REQUEST_PENDING != tmp_sync && REQUEST_COMPLETED != tmp_sync) {
```

Comment:
I think that a core premise of `ompi_request_complete()` is that it should only be invoked once for a given request.  It wasn't intended to be safe to be called multiple times -- e.g., the first time, it actually completes the request, and subsequent times it effectively does nothing.

Reply:
`ompi_request_complete()` called a second time should either fail cleanly returning error, or just happily continue as a non-op because the request already completed, but it should NEVER segfault as would happen currently the user call `MPI_Request_complete()` two times. The MPI standard does not say whether the routine can be called multiple times on the same request handle. With this fix, I'm just playing safe and allowing multiple calls to `MPI_Request_complete()`. For reference, IIRC, MPICH errors not the second but the third time, this is something I was planning to report and/or eventually fix. 

IMHO, it is probably easier to just allow multiple `MPI_Request_complete()` calls on a request, just like I'm doing here. If you do not like such semantics and prefer to error, I believe the fix is trivial.